src/PersonObjects/Sleeve/ui/TaskSelector.tsx

```diff
@@ -247,89 +240,49 @@ const canDo: {
   Synchronize: (sleeve: Sleeve) => sleeve.sync < 100,
 };
 
-function getABC(sleeve: Sleeve): [string, string, string] {
-  const work = sleeve.currentWork;
-  if (work === null) return ["Idle", "------", "------"];
-  switch (work.type) {
-    case SleeveWorkType.COMPANY:
-      return ["Work for Company", work.companyName, "------"];
-    case SleeveWorkType.FACTION: {
-      const workNames = {
-        [FactionWorkType.field]: "Field Work",
-        [FactionWorkType.hacking]: "Hacking Contracts",
-        [FactionWorkType.security]: "Security Work",
-      };
-      return ["Work for Faction", work.factionName, workNames[work.factionWorkType] ?? ""];
-    }
-    case SleeveWorkType.BLADEBURNER:
-      if (work.actionId.type === BladeburnerActionType.Contract) {
-        return ["Perform Bladeburner Actions", "Take on contracts", work.actionId.name];
-      }
-      return ["Perform Bladeburner Actions", work.actionId.name, "------"];
-    case SleeveWorkType.CLASS: {
-      if (!work.isGym()) return ["Take University Course", work.classType, work.location];
-      const gymNames: Record<GymType, string> = {
-        [GymType.strength]: "Train Strength",
-        [GymType.defense]: "Train Defense",
-        [GymType.dexterity]: "Train Dexterity",
-        [GymType.agility]: "Train Agility",
-      };
-      return ["Workout at Gym", gymNames[work.classType as GymType], work.location];
-    }
-    case SleeveWorkType.CRIME:
-      return ["Commit Crime", getEnumHelper("CrimeType").getMember(work.crimeType, { alwaysMatch: true }), "------"];
-    case SleeveWorkType.SUPPORT:
-      return ["Perform Bladeburner Actions", "Support main sleeve", "------"];
-    case SleeveWorkType.INFILTRATE:
-      return ["Perform Bladeburner Actions", "Infiltrate Synthoids", "------"];
-    case SleeveWorkType.RECOVERY:
-      return ["Shock Recovery", "------", "------"];
-    case SleeveWorkType.SYNCHRO:
-      return ["Synchronize", "------", "------"];
-  }
-}
-
 export function TaskSelector(props: IProps): React.ReactElement {
-  const abc = getABC(props.sleeve);
-  const [s0, setS0] = useState(abc[0]);
-  const [s1, setS1] = useState(abc[1]);
-  const [s2, setS2] = useState(abc[2]);
+  const s0 = props.abc[0];
+  const s1 = props.abc[1];
+  const s2 = props.abc[2];
 
-  const validActions = Object.keys(canDo).filter((k) => (canDo[k] as (sleeve: Sleeve) => boolean)(props.sleeve));
+  const validActions = Object.keys(canDo).filter((taskType) => {
+    const canDoTask = canDo[taskType];
+    if (canDoTask === undefined) {
+      return false;
+    }
+    return canDoTask(props.sleeve);
+  });
```

Comment:
```suggestion
  const validActions = Object.keys(canDo).filter((taskType) => canDo[taskType]?.(props.sleeve));
```
The optional chaining shouldn't even be needed, see if that goes away with my suggestion above.